lib/fluent/engine.rb

```diff
@@ -76,6 +76,10 @@ def parse_config(io, fname, basepath = Dir.pwd, v1_config = false)
     def run_configure(conf)
       configure(conf)
       conf.check_not_fetched { |key, e|
+        if plugin_name = e.unused_in
+          $log.warn "section <#{e.name}> is not used in #{plugin_name}."
```

Comment:
IMO, this warning should contain parent section name, ex: `is not used in <section> of #{plugin_name}`, `.. in <server> of #{plugin_name}`.